ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java

```diff
@@ -1026,6 +1029,38 @@ public void testFindNextCompactThrowsTException() throws Exception {
     verify(msc, times(0)).markFailed(any());
   }
 
+  @Test
+  public void testDoesntGatherStatsIfCompactionFails() throws Exception {
+    try (
+      MockedStatic<Worker.StatsUpdater> statsUpdater = Mockito.mockStatic(Worker.StatsUpdater.class)) {
+
+      Table t = newTable("default", "mtwb", false);
+
+      addBaseFile(t, null, 20L, 20);
+      addDeltaFile(t, null, 21L, 22L, 2);
+      addDeltaFile(t, null, 23L, 24L, 2);
+
+      burnThroughTransactions("default", "mtwb", 25);
+
+      CompactionRequest rqst = new CompactionRequest("default", "mtwb", CompactionType.MINOR);
+      String initiatorVersion = "INITIATOR_VERSION";
+      rqst.setInitiatorVersion(initiatorVersion);
+      txnHandler.compact(rqst);
+
+      Worker worker = Mockito.spy(new Worker());
+
+      doThrow(new RuntimeException()).when(worker).getMrCompactor();
+
+      worker.setConf(conf);
+      String workerVersion = "WORKER_VERSION";
+      doReturn(workerVersion).when(worker).getRuntimeVersion();
```

Comment:
no need to set the worker version, init should be enough.

Reply:
Removed, thank you.